memory/allocator.go

```diff
@@ -4,28 +4,76 @@ import (
 	"fmt"
 	"sync/atomic"
 
+	"github.com/apache/arrow/go/arrow/memory"
 	"github.com/influxdata/flux/codes"
 	"github.com/influxdata/flux/internal/errors"
 )
 
+// DefaultAllocator is the default memory allocator for Flux.
+var DefaultAllocator = memory.DefaultAllocator
+
 // Allocator tracks the amount of memory being consumed by a query.
```

Comment:
It would be good to say that this implements the Arrow allocator interface in this comment.

Reply:
Or even add a line a `var _ memory.Allocator ...` that enforces it.

Reply:
I'll add a comment, but I can't really add the above because it's already that type in the type system.